dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java

```diff
@@ -246,46 +250,44 @@ public Response renderHTMLOnly(@Context final HttpServletRequest request,
-
-            final Map<String, Object> pageMap = new HashMap(page.getMap());
-            final String templateIdentifier = (String) pageMap.get(HTMLPageAssetAPI.TEMPLATE_FIELD);
-            pageMap.remove(HTMLPageAssetAPI.TEMPLATE_FIELD);
-
-            pageMapBuilder.putAll(pageMap);
-            pageMapBuilder.putAll(getLockMap(user, page, info));
-
-            final ImmutableMap<Object, Object> templateMap = ImmutableMap.builder().put("drawed", template.isDrawed())
-                    .put("canEdit", this.permissionAPI.doesUserHavePermission(template, PermissionLevel.EDIT.getType(), user))
-                    .put("id", templateIdentifier)
-                    .build();
-
-            pageMapBuilder.put("template", templateMap);
-            pageMapBuilder.put("viewAs", createViewAsMap(request));
-
-            if(info.getLiveInode()!=null) {
-                pageMapBuilder.put("liveInode", info.getLiveInode())
-                .put("shortyLive", APILocator.getShortyAPI().shortify(info.getLiveInode()));
+            if (page != null) {
+                final Builder<String, Object> responseMapBuilder = ImmutableMap.builder();
+
+                final Host host = APILocator.getHostAPI().find(page.getHost(), user, mode.respectAnonPerms);
+                request.setAttribute(WebKeys.CURRENT_HOST, host);
+                request.getSession().setAttribute(WebKeys.CURRENT_HOST, host);
+
+                final Template template = this.templateAPI.findWorkingTemplate(page.getTemplateId(), APILocator.getUserAPI().getSystemUser(), false);
+
+                responseMapBuilder
+                        .put("html", this.pageResourceHelper.getPageRendered(page, request, response, user, mode))
+                        .put("page", this.pageResourceHelper.getPageMap(page, user))
+                        .put("containers", this.pageResourceHelper.getMappedContainers(template))
+                        .put("viewAs", createViewAsMap(request))
+                        .put("canCreateTemplate", APILocator.getLayoutAPI().doesUserHaveAccessToPortlet("templates", user));
+
+                if (template.isDrawed()) {
+                    responseMapBuilder.put("layout", DotTemplateTool.themeLayout(template.getInode()));
+                } 
+
+                if (this.permissionAPI.doesUserHavePermission(template, PermissionLevel.READ.getType(), user, false)) {
+                    responseMapBuilder.put("template",
+                        ImmutableMap.builder()
+                            .put("canEdit", this.permissionAPI.doesUserHavePermission(template, PermissionLevel.EDIT.getType(), user))
+                            .putAll(this.pageResourceHelper.asMap(template))
+                            .build());
+                }
+
+                final Response.ResponseBuilder responseBuilder = Response.ok(responseMapBuilder.build());
+                responseBuilder.header("Access-Control-Expose-Headers", "Authorization");
```

Comment:
I think we have annotations for this headers, by using RequestFilters

Reply:
Yeah, we do not need this here.  We authenticate a different way